internal/healthchecks/healthchecks.go

```diff
@@ -92,14 +121,19 @@ func HealthCheckRegistryFactory() HealthCheckRegistry {
 	}
 }
 
-func (r HealthCheckRegistry) RunAllHealthChecks(logger *log.Logger) []HealthCheckResult {
+func (r HealthCheckRegistry) RunAllHealthChecks(logger *logs.StructuredLogger) []HealthCheckResult {
 	var result []HealthCheckResult
 
 	for _, c := range r {
 		r := HealthCheckResult{Name: c.Name(), Err: c.RunCheck(logger)}
-		logger.Println(r)
+		for _, l := range r.ToStringSlice() {
```

Comment:
After thinking a while on how improve the usage of all the `if's` between `Error` and `Info` logs, my suggestion is the following (hopefully it makes sense and it's not too convoluted) :

1. Add a method `LogResult(logger *logs.StructuredLogger)` to `HealthCheckResult` like
```
func (r HealthCheckResult) LogResult(logger *logs.StructuredLogger) {
	for _, m := range r.StringSlice() {
		if r.Err != nil {
			logger.Infof(m)
		} else {
			logger.Errorf(m)
		}
        }
}
```

2. Update `LogHealthCheckResults` to use `LogResult`
```
func LogHealthCheckResults(healthCheckResults []HealthCheckResult, logger *logs.StructuredLogger) {
	for _, result := range healthCheckResults {
		result.LogResult(logger)
	}
}
```

3. Expand the definition of `StructuredLogger` to be an interface. 
```
type StructuredLogger interface {
	func Infof(format string, v ...any)
	func Errorf(format string, v ...any)
	func Println(format string, v ...any)
}
```
This interface can implemented by the `zap logger` like this (very similar to what you have right now)
```
type ZapStructuredLogger struct {
     logger *zap.SugaredLogger
}

func (f ZapStructuredLogger) Infof(format string, v ...any) {
	f.logger.Infof(format, v...)
}

func (f ZapStructuredLogger) Errorf(format string, v ...any) {
	f.logger.Errorf(format, v...)
}

func (f ZapStructuredLogger) Println(v ...any) {
	f.logger.Infoln(v...)
}
```
But also a custom `Windows Event Logger` or `Linux` could implement this interface like this 
```
windowsEventLogger := CustomStructuredLogger{func(str) { s.log.Info(ID, str)}, func(str) { s.log.Error(ID, str)}}

type CustomStructuredLogger struct {
     func info(s string)
     func error(s string)
}

func (f CustomStructuredLogger) Infof(format string, v ...any) {
	info(format)
}

func (f CustomStructuredLogger) Errorf(format string, v ...any) {
	error(format)
}

func (f CustomStructuredLogger) Println(v ...any) {
	info(format)
}
``` 

4. This way `StructuredLogger` would abstract all the complexity of any kind of logger used, not just a zap logger. This could be a step further in a having an homogenous way of logging in every place of the Ops Agent (of course, we could do this in a future PR). Also `ErrorMessages` and `InfoMessages` won't be needed anymore.